samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs

```diff
@@ -109,64 +146,327 @@ private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvok
+                        {
+                            return new MessagingExtensionActionResponse
+                            {
+                                Task = new TaskModuleContinueResponse
+                                {
+                                    Value = new TaskModuleTaskInfo
+                                    {
+                                        Card = GetAdaptiveCardAttachmentFromFile("justintimeinstallation.json"),
+                                        Height = 200,
+                                        Width = 400,
+                                        Title = "Adaptive Card - App Installation",
+                                    },
+                                },
+                            };
+                        }
+                        throw; // It's a different error.
+                    }
                     return new MessagingExtensionActionResponse
                     {
                         Task = new TaskModuleContinueResponse
                         {
                             Value = new TaskModuleTaskInfo
                             {
-                                Card = GetAdaptiveCardAttachmentFromFile("justintimeinstallation.json"),
+                                Card = GetAdaptiveCardAttachmentFromFile("adaptiveCard.json"),
                                 Height = 200,
                                 Width = 400,
-                                Title = "Adaptive Card - App Installation",
+                                Title = $"Welcome {memberName}",
                             },
                         },
                     };
-                }
-                throw; // It's a different error.
             }
+        }
 
-            return new MessagingExtensionActionResponse
+        private MessagingExtensionActionResponse razorMethod(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            var response = new MessagingExtensionActionResponse()
             {
-                Task = new TaskModuleContinueResponse
+                Task = new TaskModuleContinueResponse()
                 {
-                    Value = new TaskModuleTaskInfo
+                    Value = new TaskModuleTaskInfo()
                     {
-                        Card = GetAdaptiveCardAttachmentFromFile("adaptiveCard.json"),
-                        Height = 200,
-                        Width = 400,
-                        Title = $"Welcome {memberName}",
+                        Height = 300,
+                        Width = 300,
+                        Title = "Task Module Razor View",
+                        Url = "YourDeployedBotUrl" + "/Home/RazorView",
                     },
                 },
             };
+            return response;
         }
 
+        private MessagingExtensionActionResponse htmlMethod(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
```

Comment:
'htmlMethod' is a non-descriptive name.  Please change this to something demonstrating a more practical example (rather than 'html' and 'razor')